src/test/lib/storage/segment_access_counter_test.cpp

```diff
@@ -74,36 +74,49 @@ TEST_F(SegmentAccessCounterTest, AssignmentOperator) {
 
 TEST_F(SegmentAccessCounterTest, AccessPatternIncreasing) {
   auto positions = std::make_shared<RowIDPosList>();
-  EXPECT_EQ(AccessPattern::Point, _access_pattern(positions));
+  EXPECT_EQ(_access_pattern(positions), AccessPattern::Point);
   positions->push_back({ChunkID{0}, ChunkOffset{0}});
-  EXPECT_EQ(AccessPattern::Point, _access_pattern(positions));
+  EXPECT_EQ(_access_pattern(positions), AccessPattern::Point);
   positions->push_back({ChunkID{0}, ChunkOffset{0}});
-  EXPECT_EQ(AccessPattern::Point, _access_pattern(positions));
+  EXPECT_EQ(_access_pattern(positions), AccessPattern::Point);
   positions->push_back({ChunkID{0}, ChunkOffset{1}});
-  EXPECT_EQ(AccessPattern::SequentiallyIncreasing, _access_pattern(positions));
+  EXPECT_EQ(_access_pattern(positions), AccessPattern::SequentiallyIncreasing);
   positions->push_back({ChunkID{0}, ChunkOffset{3}});
-  EXPECT_EQ(AccessPattern::RandomlyIncreasing, _access_pattern(positions));
+  EXPECT_EQ(_access_pattern(positions), AccessPattern::RandomlyIncreasing);
   positions->push_back({ChunkID{0}, ChunkOffset{3}});
-  EXPECT_EQ(AccessPattern::RandomlyIncreasing, _access_pattern(positions));
+  EXPECT_EQ(_access_pattern(positions), AccessPattern::RandomlyIncreasing);
   positions->push_back({ChunkID{0}, ChunkOffset{1}});
-  EXPECT_EQ(AccessPattern::Random, _access_pattern(positions));
+  EXPECT_EQ(_access_pattern(positions), AccessPattern::Random);
+}
+
+TEST_F(SegmentAccessCounterTest, AccessPatternIncreasing2) {
```

Comment:
Since the number of possible state transitions is not too high, I suggest that we should test all of them. Especially the transition from `AccessPattern::Point` to `AccessPattern::RandomlyDecreasing` which is currently missing.

Reply:
Added. i made a spreadsheet and checked that all transitions are in there. No good idea how to represent that in code. Wouldn't want 30 individual test cases, especially as I also see a value in having more complex ones to test the logic beyond a single state transition.

Reply:
Also renamed RandomlyIncreasing to MonotonicallyIncreasing. The former sounds like an oxymoron.